kenjutsu/kenjutsu.py

```diff
@@ -285,6 +295,11 @@ def len_slice(a_slice, a_length=None):
             4
     """
 
+    if isinstance(a_slice, numbers.Integral):
+        raise TypeError(
+            "An integral index does not provide an object with a length."
+        )
```

Comment:
Technically `ValueError` could be appropriate here as well. This is because the value provided to `len_slice` cannot be operated on. `TypeError` is more for an actual type support issue, which this isn't exactly. Though there are already a lot of cases where `ValueError` is raised ATM. So it makes it hard to distinguish this case from others.